tests/functional/test_install_vcs.py

```diff
@@ -180,23 +180,52 @@ def test_git_with_tag_name_as_revision(script):
     assert '0.1' in version.stdout
 
 
-@pytest.mark.network
-def test_git_with_ref_as_revision(script):
+def _add_ref(script, path, ref):
     """
-    Git backend should be able to install from a ref
+    Add a new ref to a repository at the given path.
     """
-    version_pkg_path = _create_test_package(script)
-    script.run(
-        'git', 'update-ref', 'refs/foo/bar', 'HEAD',
-        expect_stderr=True,
-        cwd=version_pkg_path,
-    )
-    _change_test_package_version(script, version_pkg_path)
-    script.pip(
-        'install', '-e', '%s@refs/foo/bar#egg=version_pkg' %
-        ('git+file://' + version_pkg_path.abspath.replace('\\', '/')),
-        expect_stderr=True
-    )
+    script.run('git', 'update-ref', ref, 'HEAD', expect_stderr=True, cwd=path)
+
+
+def make_version_pkg_url(path, revision=None):
+    path = path.abspath.replace('\\', '/')
+    url_rev = '' if revision is None else '@' + revision
+    url = 'git+file://{}{}#egg=version_pkg'.format(path, url_rev)
+
+    return url
+
+
+def test_git_install_ref(script):
```

Comment:
The decorator is essentially for marking all tests that reach out to the network.

Reply:
The partitioning is so that you can run tests while not connected to the internet.

If this test runs with no internet connectivity, then it doesn't need this decorator and vice versa.

Reply:
Okay, I'll leave the decorator off then because as near as I can tell it doesn't require internet connectivity.